warehouse/integrations/bigquery/bigquery_test.go

```diff
@@ -1095,6 +1098,85 @@ func TestIntegration(t *testing.T) {
 			)
 			require.Equal(t, records, whth.SampleTestRecords())
 		})
+		t.Run("multiple files", func(t *testing.T) {
+			testCases := []struct {
+				name             string
+				loadByFolderPath bool
+			}{
+				{name: "loadByFolderPath = false", loadByFolderPath: false},
+				{name: "loadByFolderPath = true", loadByFolderPath: true},
+			}
+			for i, tc := range testCases {
+				t.Run(tc.name, func(t *testing.T) {
+					tableName := "multiple_files_test_table" + strconv.Itoa(i)
+					repeat := 10
+					loadObjectFolder := "rudder-warehouse-load-objects"
+					sourceID := "test_source_id"
+
+					prefixes := []string{loadObjectFolder, tableName, sourceID, uuid.New().String() + "-" + tableName}
+
+					loadFiles := lo.RepeatBy(repeat, func(int) whutils.LoadFile {
+						sourceFile, err := os.Open("../testdata/load.json.gz")
+						require.NoError(t, err)
+						defer func() { _ = sourceFile.Close() }()
+
+						tempFile, err := os.CreateTemp("", "clone_*.json.gz")
+						require.NoError(t, err)
+						defer func() { _ = tempFile.Close() }()
+
+						_, err = io.Copy(tempFile, sourceFile)
+						require.NoError(t, err)
+
+						f, err := os.Open(tempFile.Name())
+						require.NoError(t, err)
+						defer func() { _ = f.Close() }()
+
+						uploadOutput, err := fm.Upload(context.Background(), f, prefixes...)
+						require.NoError(t, err)
+						return whutils.LoadFile{Location: uploadOutput.Location}
+					})
+					mockUploader := newMockUploader(t, loadFiles, tableName, schemaInUpload, schemaInWarehouse)
+
+					c := config.New()
+					c.Set("Warehouse.redshift.loadByFolderPath", tc.loadByFolderPath)
```

Comment:
Warehouse.redshift -> Warehouse.bigquery

Can we add assertions to verify that the correct values or functions are being used in both cases? For instance, the current test checks only the "false" version in both scenarios.

Reply:
Yes, added an assertion based on whether it's true or false.